axonserver/src/main/java/io/axoniq/axonserver/message/event/EventStore.java

```diff
@@ -57,14 +59,51 @@
                                                             StreamObserver<Confirmation> responseObserver);
 
     /**
-     * Read events for an aggregate.
+     * Returns a {@link Flux} of all {@link SerializedEvent}s for an aggregate according whit the specified request.
+     * The events could start with a snapshot event, if the request allows the usage of the snapshots. All the events
+     * should have a sequential sequence number.
      *
-     * @param context                the context to read from
-     * @param request                the request containing the aggregate identifier and read options
-     * @param responseStreamObserver {@link StreamObserver} where the events will be published
+     * @param context        the context containing the aggregate
+     * @param authentication the authentication
+     * @param request        the request containing the aggregate identifier and read options
+     * @return a {@link Flux} of all {@link SerializedEvent}s for an aggregate according whit the specified request.
+     */
+    Flux<SerializedEvent> aggregateEvents(String context,
+                                          Authentication authentication,
+                                          GetAggregateEventsRequest request);
+
+    /**
+     * Returns a {@link Flux} of all snapshots events for an aggregate according whit the specified request. The
+     * snapshots are sorted from the latest one. The snapshot sequence numbers are not sequential.
+     *
+     * @param context        the context containing the aggregate
+     * @param authentication the authentication
+     * @param request        the request containing the aggregate identifier and read options
+     * @return a {@link Flux} of all {@link SerializedEvent}s for an aggregate according whit the specified request.
      */
-    void listAggregateEvents(String context, Authentication authentication, GetAggregateEventsRequest request,
-                             StreamObserver<SerializedEvent> responseStreamObserver);
+    default Flux<SerializedEvent> aggregateSnapshots(String context,
+                                                     Authentication authentication,
+                                                     GetAggregateSnapshotsRequest request) {
+        return Flux.create(
+                sink -> listAggregateSnapshots(context, authentication, request, new StreamObserver<SerializedEvent>() {
+                    @Override
+                    public void onNext(SerializedEvent serializedEvent) {
+                        if (serializedEvent.getAggregateSequenceNumber() < request.getMaxSequence()) {
```

Comment:
it should be the responsibility of the listAggregateSnapshots operation to ensure that the sequence number is in the valid range.